saleor/graphql/product/tests/queries/test_product_variants_query_with_where.py

```diff
@@ -67,3 +70,98 @@ def test_product_variant_filter_by_ids_empty_list(
     data = get_graphql_content(response)
     variants = data["data"]["productVariants"]["edges"]
     assert len(variants) == 0
+
+
+@pytest.mark.parametrize(
+    "stocks_to_create",
+    [[], [("1", 0), ("2", 1), ("4", 1)]],
+)
+def test_product_variant_where_in_stock(
+    api_client,
+    product_variant_list,
+    channel_USD,
+    stocks_to_create,
+    shipping_zones_with_warehouses,
+):
+    # given
+    warehouse = shipping_zones_with_warehouses[0].warehouses.first()
+    channel_USD.warehouses.add(warehouse)
+    channel_USD.save()
+    Stock.objects.all().delete()
+    Stock.objects.bulk_create(
+        [
+            Stock(
+                product_variant=ProductVariant.objects.filter(sku=sku).first(),
+                quantity=quantity,
+                warehouse=warehouse,
+            )
+            for sku, quantity in stocks_to_create
+        ]
+    )
+    variables = {
+        "channel": channel_USD.slug,
+        "where": {"stockAvailability": "IN_STOCK"},
+    }
+
+    # when
+    response = api_client.post_graphql(PRODUCT_VARIANTS_WHERE_QUERY, variables)
+
+    # then
+    data = get_graphql_content(response)
+    variants = data["data"]["productVariants"]["edges"]
+
+    positive_stock_variants = [
+        sku for sku, quantity in stocks_to_create if quantity > 0
+    ]
+    assert [variant["node"]["sku"] for variant in variants] == positive_stock_variants
+
+
+@pytest.mark.parametrize(
+    "stocks_to_create",
+    [[], [("1", 0), ("2", 1), ("4", 1)]],
+)
+def test_product_variant_where_out_of_stock(
+    api_client,
+    product_variant_list,
+    channel_USD,
+    stocks_to_create,
+    shipping_zones_with_warehouses,
+):
+    # given
+    warehouse = shipping_zones_with_warehouses[0].warehouses.first()
+    channel_USD.warehouses.add(warehouse)
+    channel_USD.save()
```

Comment:
This save is unnecessary, calling `.add()` on a Django model relation is sufficient to save the relationship in the database.